### PR TITLE
Run jdk9+ tests

### PR DESCRIPTION
**Pull request by raboof**
Manually checked that indeed both scala and java jdk9+only tests are
now ran when testing on jdk9.

I think we should add a nightly validation check that does some
smoke-checking on the validation results, but that's for another PR

### CODE REVIEW

**Review by johanandren — APPROVED** [review]
LGTM. Thanks a bunch for looking into it!

**Comment by johanandren** [comment]
I cherry picked this into the JFR branch, but with a syntax error in one of the JDK9+ sources and running `akka-remote/test` on Java 11 I still don't get a compilation error. Plus that the test isn't run ofc. Even with `akka-remote/TestJdk9/test` I don't get any compilation or testing to happen.

**Comment by johanandren** [comment]
Ok, mystery half solved, had named the directory wrong `scala-jdk-9-only`, so that's why the test wasn't found. Running test doesn't seem trigger compilation of the jdk9 main sources though so I think I'd have to manually first do a `CompileJdk9/compile` to get the tests to actually run against the latest jdk9 main class

**Comment by johanandren** [comment]
Next discovery: when explicitly running `akka-remote/CompileJdk9/compile` and then `akka-remote/TestJdk9/test` it compiles the sources and test sources but then doesn't actually find the class when running the test.

**Comment by johanandren** [comment]
If I make the TestJdk9 compile depend on the `CompileJdk9/compile` that seems to do the trick for triggering compilation as expected: 

```
compile := compile.dependsOn(CompileJdk9 / compile).value,
```

**Comment by raboof** [comment]
but still doesn't find the tests? do the jdk9+ tests in `akka-stream-tests` get found for you?

**Comment by johanandren** [comment]
Ah, the difference is probably that it is in a separate module vs tests in the same module for me.

**Comment by raboof** [comment]
After we merge this we should add the `checkTestsHaveRun` target to the nightly build and switch it to JDK11

**Review by patriknw — APPROVED** [review]
LGTM, if you say so

**Comment by raboof** [comment]
(these changes were already pulled in by #28305)

**Pull request closed without merging**
